mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -325,6 +326,22 @@ def GPU_GPUFuncOp : GPU_Op<"func", [
 
     /// Verifies the body of the function.
     LogicalResult verifyBody();
+
+    //===------------------------------------------------------------------===//
+    // CallableOpInterface
+    //===------------------------------------------------------------------===//
+
+    /// Returns the region on the current operation that is callable. This may
+    /// return null in the case of an external callable object, e.g. an external
+    /// function.
+    ::mlir::Region *getCallableRegion() { return isExternal() ? nullptr
```

Comment:
We want to allow inlining into a gpu.func.
Would this change makes gpu.func inlinable to a function?
I am thinking that we probably don't want to inline gpu.func to another function, given that the gpu.func stores information like workgroup, private vars, etc.
Should we always return nullptr to invalid inlining gpu.func to another function?

Reply:
This change allows a `gpu.func` operation to be considered as a node in the call graph. This is needed in order to allow inlining call edges stemming from a `gpu.func` operation.

Reply:
Right, but I wonder if there is a side effort, which also inline gpu.func body to another function.

Reply:
I see that you added code to disallow gpu.func inline to gpu.func. Is there anything stopping gpu.func inline to e.g., func.func?

Reply:
No there isn't (assuming the callee func.func operation doen't contain ops preventing inlining).

Reply:
Do we want gpu.func inline to func.func? I worry that there will be information lose from doing so, like workgroup and private var infos.

Reply:
I agree with Whitney: inlining a `gpu.func` (kernel) into a `func.func` would in fact change the semantics of the program, as no offloading to the SYCL device would happen.

Reply:
In practice this will not happen because our `gpu.func` are kernels and they aren't called by any function in the gpu.module.

Reply:
In principle though I agree. Turns out that currently cgeist 'barfs' if a func.func calls a gpu.func.  If you guys are OK, I'd like to defer investigation and prevention of this happening in a future PR, and land this one.

Reply:
Actually I have modified the code to prevent a gpu.func to be inlined in a func.func (or another gpu.func).